Updated DataGrid FIltering topics (#837)
telerik/maui-docs: controls/datagrid/filtering/filter-control-template.md

* `FilterControlTemplate`(`DataTemplate`)&mdash;Specifies the user defined template used for Filtering UI. The template must contain an instance of the `Telerik.Maui.Controls.Compatibility.DataGrid.DataGridFilterControlBase` class. 

> The DataGrid provides a few predefined filter controls which are applied depending on the column type:

Reviewer: ```suggestion
The DataGrid provides the following predefined filter controls which are applied depending on the column type:
```

* `DataGridTextFilterControl`
* `DataGridNumericFilterControl`
* `DataGridBooleanFilterControl`
* `DataGridDateTimeFilterControl`
* `DataGridTimeFilterControl`
* `DataGridComboBoxFilterControl`

The `FilterControlTemplate` property of the columns allows you to customize its corresponding filter control. If the default filter control does not meet the requirements, you can create a custom filter control from scratch. 

> ## Example with modified TextFilterControl

Reviewer: ```suggestion
## Example with Modified TextFilterControl
```

> The example below shows how to apply the `DataGridTextFilterControl` as a `FilterControlTemplate` of the text column and set the filter match case options to `False`. This means the toggle buttons inside the filter template for switching to case-sensitive filter will be untoggled when the filtering UI appears.

Reviewer: ```suggestion
The example below shows how to apply the `DataGridTextFilterControl` as a `FilterControlTemplate` of the text column and set the filter match case options to `False`. This means that the toggle buttons inside the filter template for switching to case-sensitive filter will be untoggled when the filtering UI appears.
```
